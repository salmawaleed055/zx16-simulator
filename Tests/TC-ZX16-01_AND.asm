li x1, 7
li x2, 2
and x1, x1, x2
ecall 0x3FF