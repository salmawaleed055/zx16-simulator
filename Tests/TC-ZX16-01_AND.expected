0x0000: 0E79    li x1, 7
0x0002: 04B9    li x2, 2
0x0004: 1440    add x1, x1, x2  # Assuming 'add x1, x1, x2' is encoded as 1440 if it's a pseudo-instruction for 'add rd, rs1, rs2'
0x0006: FFC7    ecall 0x3FF

--- Final State ---
x0: 0x0000
x1: 0x0002
x2: 0x0002
x3: 0x0000
x4: 0x0000
x5: 0x0000
a0: 0x0000
a1: 0x0000
PC: 0x0008
---------------------
Simulation finished.